scripts/install_backend_python_libs.py

```diff
@@ -29,19 +30,53 @@
 
 
 def _normalize_python_library_name(library_name):
-    """Returns a normalized (lowercase) version of the python library name.
-
-    Python library name strings are case insensitive which means that
-    libraries are equivalent even if the casing of the library names are
-    different. This function normalizes python library names such that
-    identical libraries have the same library name strings.
+    """Returns a normalized version of the python library name.
+
+    Normalization of a library name means converting the library name to
+    lowercase, and removing any "[...]" suffixes that occur. The reason we do
+    this is because of 2 potential confusions when comparing library names that
+    will cause this script to find incorrect mismatches.
+
+    1. Python library name strings are case insensitive which means that
+       libraries are equivalent even if the casing of the library names are
+       different.
+    2. There are certain python libraries with a default version and multiple
+       variants. These variants have names like `library[sub-library]` and
+       signify that it is a version of the library with special support for
+       the sub-library.
+
+    Here are some examples of ambiguities that this function resolves:
+    - 'googleappenginemapreduce' is listed in the 'requirements.txt' file as
+      all lowercase. However, the installed directories has names starting with
+      the string 'GoogleAppEngineMapReduce'. This causes confusion when
+      searching for mismatches because python treats the two library names as
+      different even though they are equivalent.
+    - The name 'google-api-core[grpc]', listed in a 'requirements.txt' file
```

Comment:
If the name 'google-api-core[grpc]' is listed in a 'requirements.txt' file, this means that ...

Reply:
Done.